[PY3] Add unicode_literals to high-priority modules/states

terminalmage opened the pull request:
Remove this section if not relevant

Remove this section if not relevant


Yes/No


Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

terminalmage committed:
[PY3] Add unicode_literals to high-priority modules/states

rallytime commented:
@terminalmage There a some test failures here that are related. I think these ones are for sure related, the others might need another run to see if it's flakiness or related:

- integration.modules.test_cp.CPModuleTest.test_get_file_str_https
- integration.modules.test_cp.CPModuleTest.test_get_file_str_local
- integration.modules.test_cp.CPModuleTest.test_get_url_file_no_dest
- integration.modules.test_cp.CPModuleTest.test_get_url_https_no_dest

terminalmage commented:
The cp module tests were failing for the same reason as some of the others, because of str/unicode mismatches. I added my same workaround for ``ModuleCase.run_function()`` that I added for the git module integration tests.

I could not reproduce the other failures, and did not touch the runners tests anyway.

rallytime commented:
re-run py

rallytime commented on tests/unit/modules/test_disk.py:
```diff
@@ -141,7 +141,7 @@ def test_tune(self):
                 self.assertEqual(kwargs, {'python_shell': False})
 
-    def test_format(self):
+    def test_ormat(self):
```
I think this was an accident to change the name of this test function.

rallytime reviewed (changes requested):
Only one small change, otherwise this looks great.

rallytime commented:
@terminalmage This has a merge conflict now, as well.

terminalmage commented:
@rallytime This is fixed and rebased.
